### PR TITLE
Correctly handle IntelliJ runs for root project names with spaces

### PR DESCRIPTION
**Pull request by Technici4n**

### CODE REVIEW

File: src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java [hunk 1]
```diff
@@ -299,6 +300,27 @@ private static String evaluateXPath(File file, @Language("xpath") String express
...
+     * Do not use {@link ModuleRef} as it does not correctly handle projects with a space in their name!
+     */
+    public static String getIntellijModuleName(Project project, SourceSet sourceSet) {
+        var moduleName = new StringBuilder();
+        // The `replace` call here is our bug fix compared to ModuleRef!
+        // The actual IDEA logic is more complicated, but this should cover the majority of use cases.
+        // See https://github.com/JetBrains/intellij-community/blob/a32fd0c588a6da11fd6d5d2fb0362308da3206f3/plugins/gradle/src/org/jetbrains/plugins/gradle/service/project/GradleProjectResolverUtil.java#L205
+        // which calls https://github.com/JetBrains/intellij-community/blob/a32fd0c588a6da11fd6d5d2fb0362308da3206f3/platform/util-rt/src/com/intellij/util/PathUtilRt.java#L120
+        moduleName.append(project.getRootProject().getName().replace(" ", "_"));
+        if (!project.getPath().equals(":")) {
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java:315]: I'd check if the rootProject == the project instead.

File: src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java [hunk 2]
```diff
@@ -299,6 +300,27 @@ private static String evaluateXPath(File file, @Language("xpath") String express
...
+            moduleName.append(project.getPath().replaceAll(":", "."));
+        }
+        if (!sourceSet.getName().isEmpty()) {
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java:318]: Can a sourceset name be empty or blank?

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java (hunk 1 above)
Comment (Technici4n) [reply src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java:315]: We could. This just mimics the logic inside idea-ext.

File: src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java (hunk 2 above)
Comment (Technici4n) [reply src/main/java/net/neoforged/moddevgradle/internal/RunUtils.java:318]: It's suspicious but I took it from idea-ext. :D

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**